processing/src/main/java/org/apache/carbondata/processing/csvreaderstep/UnivocityCsvParser.java

```diff
@@ -104,6 +108,20 @@ public void initialize() throws IOException {
   }
 
   /**
+   * This method will decide the number of coulmns to be parsed for a row by univocity parser
+   *
+   * @param columnCountInSchema total number of columns in schema
+   * @return
+   */
+  private int getMaxColumnsForParsing(int columnCountInSchema) {
+    int maxNumberOfColumnsForParsing = columnCountInSchema;
+    if (columnCountInSchema < MAX_NUMBER_OF_COLUMNS_FOR_PARSING) {
+      maxNumberOfColumnsForParsing = MAX_NUMBER_OF_COLUMNS_FOR_PARSING;
+    }
+    return maxNumberOfColumnsForParsing;
```

Comment:
I added +10 because to avoid this bug, i think now no need to add 10 as we are allowing user to give max number of columns. @gvramana Please comment